Scripts/RemediationScripts/Remediate-SetSQLServerMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,999 @@
+                $sqlServersRemediated | Format-Table -Property $colsProperty -Wrap
+
+                # Write this to a file.
+                $sqlServersRemediatedFile = "$($backupFolderPath)\RemediatedsqlServersFileforMinTLS.csv"
+                $sqlServersRemediated| Export-CSV -Path $sqlServersRemediatedFile -NoTypeInformation
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "This information has been saved to [$($sqlServersRemediatedFile)]"
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "Use this file for any roll back that may be required." -ForegroundColor $([Constants]::MessageType.Info)
+                Write-Host $([Constants]::SingleDashLine)
+            }
+
+            if ($($sqlServersSkipped | Measure-Object).Count -gt 0)
+            {
+                Write-Host "Error changing minimal TLS version for following SQL Server(s):" -ForegroundColor $([Constants]::MessageType.Error)
+                $sqlServersSkipped | Format-Table -Property $colsProperty -Wrap
+                Write-Host $([Constants]::SingleDashLine)
+                # Write this to a file.
+                $sqlServerSkippedFile = "$($backupFolderPath)\SkippedsqlServersFileforMinTLS.csv"
+                $sqlServersSkipped | Export-CSV -Path $sqlServerSkippedFile -NoTypeInformation
+                Write-Host $([Constants]::SingleDashLine)
+                Write-Host "This information has been saved to [$($sqlServerResourcesSkippedFile)]"
+                Write-Host $([Constants]::SingleDashLine)
+            }
+        }
+
+        if($AutoRemediation)
+        {
+            $logFile = "LogFiles\"+ $($TimeStamp) + "\log_" + $($SubscriptionId) +".json"
+            $log =  Get-content -Raw -path $logFile | ConvertFrom-Json
+            foreach($logControl in $log.ControlList){
+                if($logControl.ControlId -eq $controlIds){
+                    $logControl.RemediatedResources=$logRemediatedResources
+                    $logControl.SkippedResources=$logSkippedResources
+                    $logControl.RollbackFile = $sqlServersRemediatedFile
+                }
+            }
+            
+            $log | ConvertTo-json -depth 10  | Out-File $logFile
+        }
+    }
+    else
+    {
+        Write-Host "[Step 5 of 5] Changing minimal TLS version for SQL Servers(s)"
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Skipped as -DryRun switch is provided." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::SingleDashLine)
+
+        Write-Host "**Next steps:**" -ForegroundColor $([Constants]::MessageType.Info)
+        Write-Host $([Constants]::SingleDashLine)
+        Write-Host "Run the same command with -FilePath [$($backupFile)] and without -DryRun, to change the minimal TLS version to required TLS version for all SQL Server(s) listed in the file." -ForegroundColor $([Constants]::MessageType.Info)
```

Comment:
remove bracket in here (special case) as user may directly copy paste and this will interfere with it.

Reply:
removed